feat(issues): Add more `is:for_review` details (#11023)
getsentry/sentry-docs: docs/product/issues/index.mdx

From the **Issues** page, you can begin to triage. The page is organized into tabs, each corresponding to a filtered list of issues, and these different lists help you with triaging:

- All Unresolved (`is:unresolved`): All unresolved issues, including issues that need review.

> - For Review (`is:unresolved is:for_review`): Also called **Review List**. Issues that need to be reviewed; for-review issues are a subset of all unresolved issues. These issues are new, regressions, or unresolved issues that have not yet been reviewed.

Reviewer: ```suggestion
- For Review (`is:unresolved is:for_review`). Also called **Review List**,  for-review issues are a subset of all unresolved issues and can include new issues or regressions that haven't been reviewed yet.
```
